### DISCUSSION TITLE
fail to search on all shard leaders

### DISCUSSION THREAD

**Discussion by zhenzi0322** [milvus-io/milvus#19593] [Ideas & Feature requests]
相关版本：
```
pymilvus: 2.0.2
milvus：v2.1.4【独立】
```

我在进行查询`id`和通过向量查询图片时出现如下的错误：

> `pymilvus.client.exceptions.BaseException: <BaseException: (code=1, message=fail to search on all shard leaders, err=fail to Query, QueryNode ID = 97, reason=ShardCluster for by-dev-rootcoord-dml_0_436320687559802881v0 replicaID 436320567889756162 is no available)>
`

示例：
```python
from pymilvus import Collection, connections

def main():
    connections.connect(host="192.168.0.110", port="19530")
    collection = Collection("test")  # Get an existing collection.
    res = collection.query(
        expr="id in [8]"
    )
    print(res)

if __name__ == '__main__':
    main()
```

**Comment by yhmo** [comment]
Link to https://github.com/milvus-io/milvus/issues/18956
Please collect information to help diagnose, following the comment: https://github.com/milvus-io/milvus/issues/18956#issuecomment-1233627155
